src/test/euler/staking/StakeableEulerERC4626.t.sol

```diff
@@ -206,6 +207,30 @@ contract StakeableEulerERC4626Test is Test {
         assertEq(vault.totalAssets(), deposited);
     }
 
+    function testWithdrawWithStaking() public withStakingContract {
+        uint256 deposited = 10000; // underlying + eToken (since starts 1:1)
+        uint256 staked = 1000; // eToken
+        _deposit(alice, deposited);
+        _stake(staked);
+        uint256 remaining = deposited - staked;
+
+        uint256 balanceAlice = underlying.balanceOf(alice);
+
+        // Withdraw enough that unstaking is necessary
+        uint256 withdrawFromStaked = staked / 4;
+        uint256 toWithdraw = remaining + withdrawFromStaked;
+        vm.prank(alice);
+        vault.withdraw(toWithdraw, alice, alice);
+
+        assertEq(underlying.balanceOf(alice), balanceAlice + toWithdraw);
+
+        // There should still be some staked balance
+        uint256 extraWithdrew = FixedPointMathLib.mulDivUp(staked - withdrawFromStaked, 5, 100);
+        uint256 leftStaking = staked - withdrawFromStaked - extraWithdrew;
+        assertEq(eToken.balanceOf(address(vault)), extraWithdrew);
+        assertEq(eToken.balanceOf(address(stakingRewards)), leftStaking);
+        assertEq(stakingRewards.balanceOf(address(vault)), leftStaking);
+    }
```

Comment:
Maybe we should add the case in which `eTokenStakedBalance < withdrawAttempt = true` since I don't think it's possible? Also, what happens when we withdraw all the staked balance? Although it sounds more of a e2e than a unit. But there is something about zero-ing values that we should test here imho.

Reply:
Not sure how to get `eTokenStakedBalance < withdrawAttempt = true`, but added test that withdraws all staked balance